src/app/roles/page.tsx

```diff
@@ -1,30 +1,45 @@
+"use client";
+
+import { useState } from "react";
 import HeaderLayout from "~/components/header-layout";
-import { RoleReviewCard } from "~/components/role-review-card";
+import { ReviewCard } from "~/components/review-card";
+import { ReviewCardPreview } from "~/components/review-card-preview";
 import SearchFilter from "~/components/search-filter";
-import { api } from "~/trpc/server";
-import { unstable_noStore as noStore } from "next/cache";
+import { cn } from "~/lib/utils";
+import { api } from "~/trpc/react";
 
-export default async function Roles() {
-  /**
-   * FIXME: This is a temporary fix, figure out how to get build command working without noStore();
-   * @returns A promise containing the roles from the database
-   */
-  async function getRoles() {
-    noStore();
-    const roles = await api.role.list.query();
-    return roles;
-  }
+export default function Roles() {
+  const reviews = api.review.list.useQuery();
 
-  const roles = await getRoles();
+  const [selectedReviewIdx, setSelectedReviewIdx] = useState<number>(0);
 
   return (
     <HeaderLayout>
       <SearchFilter />
-      <div className="grid grid-cols-1 gap-4 md:grid-cols-2 xl:grid-cols-3">
-        {roles.map((role) => {
-          return <RoleReviewCard key={role.id} roleObj={role} />;
-        })}
-      </div>
+      {/* TODO: Loading animations */}
+      {reviews.data && (
+        <div className="mb-8 grid w-4/5 grid-cols-5 gap-4 lg:w-3/4">
+          <div className="col-span-2 gap-3">
+            {reviews.data.map((review, idx) => {
+              return (
+                <div key={review.id} onClick={() => setSelectedReviewIdx(idx)}>
```

Comment:
Do we want to store the index of the role being selected or the review object itself?